sdk/template/template/test/public/configurationClient.spec.ts

```diff
@@ -31,13 +33,17 @@ function createConfigurationClient(): ConfigurationClient {
   // we used DefaultAzureCredential, they might be different on some machines
   // than on others, depending on which credentials are available (such as
   // Managed Identity or developer credentials).
-  const credential = new ClientSecretCredential(
-    env.AZURE_TENANT_ID,
-    env.AZURE_CLIENT_ID,
-    env.AZURE_CLIENT_SECRET
-  );
-
-  return new ConfigurationClient(endpoint, credential);
+  let credential: TokenCredential;
+  if (isPlaybackMode()) {
+    credential = new NoOpCredential();
+  } else {
+    credential = new ClientSecretCredential(
+      env.AZURE_TENANT_ID,
+      env.AZURE_CLIENT_ID,
+      env.AZURE_CLIENT_SECRET
+    );
+  }
```

Comment:
If possible it would be very nice if we could wrap this up into a helper function/credential/something in one of the following places:

- the recorder itself
- @azure/test-utils
- an identity extension (probably overkill)

Reply:
Putting it in the recorder itself or in the test-utils would mean that identity cannot depend on them.
Which means it has to be `@azure/test-utils-identity`, fine?

Reply:
(seems a bit overkill)

Reply:
New unpublished package for this

Reply:
I second @willmtemple suggestion, having a helper for this would be very nice as it will reduce the boilerplate we have to write on every package

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/19423